### PR TITLE
gitserver: add some git service debugging facilities

### PR DESCRIPTION
**Pull request by tsenart**
This commit adds a SRC_GIT_SERVICE_MAX_EGRESS_BYTES_PER_SECOND env
variable and some logging of stderr of gitserver's service commands.

This is to aid in debugging https://github.com/sourcegraph/customer/issues/586

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ All notable changes to Sourcegraph are documented in this file.
 
 - Individual batch changes can publish multiple changesets to the same repository by specifying multiple target branches using the [`on.branches`](https://docs.sourcegraph.com/batch_changes/references/batch_spec_yaml_reference#on-repository) attribute. [#25228](https://github.com/sourcegraph/sourcegraph/issues/25228)
 - Low resource overlay added. NOTE: this is designed for internal-use only. Customers can use the `minikube` overlay to achieve similar results.[#4012](https://github.com/sourcegraph/deploy-sourcegraph/pull/4012)
+- The `SRC_GIT_SERVICE_MAX_EGRESS_BYTES_PER_SECOND` environment variable to control the egress throughput of gitserver's git servce (e.g. used by zoekt-index-server to clone repos to index). Set to -1 for no limit.
```
Comment (tsenart) [inline CHANGELOG.md:20]:
```suggestion
- The `SRC_GIT_SERVICE_MAX_EGRESS_BYTES_PER_SECOND` environment variable to control the egress throughput of gitserver's git servce (e.g. used by zoekt-index-server to clone repos to index). Set to -1 for no limit. [#29197](https://github.com/sourcegraph/sourcegraph/pull/29197)
```

**Review by DaedalusG — APPROVED** [review]
Looks great, thanks for cutting this on a Friday night ❤️

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ All notable changes to Sourcegraph are documented in this file.
 
 - Individual batch changes can publish multiple changesets to the same repository by specifying multiple target branches using the [`on.branches`](https://docs.sourcegraph.com/batch_changes/references/batch_spec_yaml_reference#on-repository) attribute. [#25228](https://github.com/sourcegraph/sourcegraph/issues/25228)
 - Low resource overlay added. NOTE: this is designed for internal-use only. Customers can use the `minikube` overlay to achieve similar results.[#4012](https://github.com/sourcegraph/deploy-sourcegraph/pull/4012)
+- The `SRC_GIT_SERVICE_MAX_EGRESS_BYTES_PER_SECOND` environment variable to control the egress throughput of gitserver's git servce (e.g. used by zoekt-index-server to clone repos to index). Set to -1 for no limit. [#29197](https://github.com/sourcegraph/sourcegraph/pull/29197)
```
Comment (tsenart) [inline CHANGELOG.md:20]:
```suggestion
- The `SRC_GIT_SERVICE_MAX_EGRESS_BYTES_PER_SECOND` environment variable to control the egress throughput of gitserver's git service (e.g. used by zoekt-index-server to clone repos to index). Set to -1 for no limit. [#29197](https://github.com/sourcegraph/sourcegraph/pull/29197)
```

**Review by ggilmore — APPROVED** [review]
LGTM (discussed over slack)

**Comment by tsenart** [comment]
@kevinwojo We need this to land in the release please.

**Pull request merged**